jme3-xr/src/main/java/com/jme3/system/lwjgl/openxr/HelloOpenXRGL.java

```diff
@@ -0,0 +1,886 @@
+
+            XrCompositionLayerProjectionView projectionLayerView = projectionLayerViews.get(viewIndex)
+                .pose(views.get(viewIndex).pose())
+                .fov(views.get(viewIndex).fov())
+                .subImage(si -> si
+                    .swapchain(viewSwapchain.handle)
+                    .imageRect(rect -> rect
+                        .offset(offset -> offset
+                            .x(0)
+                            .y(0))
+                        .extent(extent -> extent
+                            .width(viewSwapchain.width)
+                            .height(viewSwapchain.height)
+                        )));
+
+            OpenGLRenderView(projectionLayerView, viewSwapchain.images.get(swapchainImageIndex), viewIndex);
+
+            check(xrReleaseSwapchainImage(
+                viewSwapchain.handle,
+                XrSwapchainImageReleaseInfo.calloc(stack)
+                    .type$Default()
+            ));
+        }
+
+        layer.space(xrAppSpace);
+        layer.views(projectionLayerViews);
+        return true;
+    }
+
+    private static Matrix4f projectionMatrix = new Matrix4f(); //TODO: Do we need this?
+    private static Matrix4f viewMatrix       = new Matrix4f(); //TODO: Do we need this?
+    private static com.jme3.math.Vector3f viewPos = new com.jme3.math.Vector3f();
+    private static com.jme3.math.Quaternion viewRot = new com.jme3.math.Quaternion();
+
+    private void OpenGLRenderView(XrCompositionLayerProjectionView layerView, XrSwapchainImageOpenGLKHR swapchainImage, int viewIndex) {
+        glBindFramebuffer(GL_FRAMEBUFFER, swapchainFramebuffer);
+
+        glFramebufferTexture2D(GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0, GL_TEXTURE_2D, swapchainImage.image(), 0);
+        glFramebufferTexture2D(GL_FRAMEBUFFER, GL_DEPTH_ATTACHMENT, GL_TEXTURE_2D, depthTextures.get(swapchainImage), 0);
+
+        XrRect2Di imageRect = layerView.subImage().imageRect();
+        glViewport(
+            imageRect.offset().x(),
+            imageRect.offset().y(),
+            imageRect.extent().width(),
+            imageRect.extent().height()
+        );
+        glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT | GL_STENCIL_BUFFER_BIT);
+
+        glFrontFace(GL_CW);
+        glCullFace(GL_BACK);
+        glEnable(GL_DEPTH_TEST);
+
+        XrPosef       pose        = layerView.pose();
+        XrVector3f    pos         = pose.position$();
+        XrQuaternionf orientation = pose.orientation();
+        XRHelper.applyProjectionToMatrix(projectionMatrix.identity(), layerView.fov(), 0.1f, 100f, false);
+        viewMatrix.translationRotateScaleInvert(
+            pos.x(), pos.y(), pos.z(),
+            orientation.x(), orientation.y(), orientation.z(), orientation.w(),
+            1, 1, 1
+        );
+        viewPos.set(pos.x(), pos.y(), pos.z());
+        viewRot.set(orientation.x(), orientation.y(), orientation.z(), orientation.w());
+        xrHmd.onUpdateHmdOrientation(viewPos, viewRot);
```

Comment:
Yeah, weirdly the field of view reported by layerView.fov().angleDown() + layerView.fov().angleUp() is around 100 degrees. But I need to set the field of view to 130 degrees for it to feel right

Reply:
I think it might be because the projection matrix is wrong. In the JME openVR code I can see that the cameras projection matrix is changed (in LWJGLOpenVRViewManager) based on what OpenVR says it should be. I wonder if something similar is needed for OpenXR